plugins/plugin-maven/che-plugin-maven-generator-archetype/src/main/java/org/eclipse/che/plugin/maven/generator/archetype/MavenArchetypeJsonRpcMessenger.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.maven.generator.archetype;
+
+import com.google.inject.Singleton;
+
+import org.eclipse.che.api.core.jsonrpc.commons.RequestHandlerConfigurator;
+import org.eclipse.che.api.core.jsonrpc.commons.RequestTransmitter;
+import org.eclipse.che.api.core.notification.EventService;
+import org.eclipse.che.api.core.notification.EventSubscriber;
+import org.eclipse.che.dto.server.DtoFactory;
+import org.eclipse.che.plugin.maven.shared.dto.ArchetypeOutput;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newConcurrentHashSet;
+import static org.eclipse.che.plugin.maven.shared.MavenAttributes.MAVEN_ARCHETYPE_CHANEL_OUTPUT;
+import static org.eclipse.che.plugin.maven.shared.MavenAttributes.MAVEN_ARCHETYPE_CHANEL_SUBSCRIBE;
+
+/**
+ * Mechanism which sends events of maven archetype generation by using JSON RPC to the client.
+ */
+@Singleton
+public class MavenArchetypeJsonRpcMessenger implements EventSubscriber<ArchetypeOutput> {
+    private final Set<String> endpointIds = newConcurrentHashSet();
+    private EventService       eventService;
+    private RequestTransmitter transmitter;
+
+    @Inject
+    public MavenArchetypeJsonRpcMessenger(EventService eventService, RequestTransmitter transmitter) {
+        this.eventService = eventService;
+        this.transmitter = transmitter;
+    }
+
+    @PostConstruct
+    private void subscribe() {
+        eventService.subscribe(this);
+    }
+
+    @PreDestroy
+    private void unsubscribe() {
+        eventService.unsubscribe(this);
+    }
+
+    @Inject
+    private void configureSubscribeHandler(RequestHandlerConfigurator configurator) {
```

Comment:
It would be nice if we could have also a method to cancel subscription here, what do you think?